clar2wasm/src/words/equal.rs

```diff
@@ -39,26 +39,31 @@ impl Word for IsEq {
         }
         val_locals.reverse();
 
-        // Traverse the second operand pushing it onto the stack.
-        let sec_op = args.get_expr(1)?;
-        generator.traverse_expr(builder, sec_op)?;
+        for val in &val_locals {
+            builder.local_get(*val);
+        }
+
+        if args.len() == 1 {
+            // push same first operand to stack
+            for val in &val_locals {
+                builder.local_get(*val);
+            }
+        } else if args.len() >= 2 {
+            // Traverse the second operand pushing it onto the stack.
+            let sec_op = args.get_expr(1)?;
+            generator.traverse_expr(builder, sec_op)?;
+        }
 
         // Equals expression needs to handle different types.
         // is-eq-int function can be reused to both int and uint types.
```

Comment:
I still believe that this comment should be right above the match arm for int/uint

Reply:
Changed.